### PR TITLE
Added Debug/GPIO14 Buttons

### PR DESCRIPTION
**Pull request by RicBent**
Added the debug button available from the development unit debugger. Only configurable from ini files.
Edit: Also added the GPIO14 button (https://www.3dbrew.org/wiki/GPIO_Services#GPIO_Bits)

Closes https://github.com/citra-emu/citra/issues/4536

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4537)

### CODE REVIEW

File: src/core/settings.h [hunk 1]
```diff
@@ -38,6 +38,7 @@ enum Values {
     R,
     Start,
     Select,
+    Dbg,
```
Comment (DarkLordZach) [inline src/core/settings.h]: Is there any reason to not just use 'Debug'? I'm not going to comment everywhere, but this applies to all the other usages too.

Comment (RicBent) [reply src/core/settings.h]: No, not really. If wanted I can change it.

File: src/citra_qt/configuration/configure_input.cpp [hunk 2]
```diff
@@ -102,7 +102,8 @@ ConfigureInput::ConfigureInput(QWidget* parent)
     button_map = {
         ui->buttonA,        ui->buttonB,        ui->buttonX,         ui->buttonY,  ui->buttonDpadUp,
         ui->buttonDpadDown, ui->buttonDpadLeft, ui->buttonDpadRight, ui->buttonL,  ui->buttonR,
-        ui->buttonStart,    ui->buttonSelect,   ui->buttonZL,        ui->buttonZR, ui->buttonHome,
+        ui->buttonStart,    ui->buttonSelect,   nullptr,             ui->buttonZL, ui->buttonZR,
```
Comment (wwylele) [inline src/citra_qt/configuration/configure_input.cpp]: `button_map[button_id]->setContextMenuPolicy(Qt::CustomContextMenu);` below will crash due to this `nullptr`

Comment (RicBent) [reply src/citra_qt/configuration/configure_input.cpp]: This is not true. The dialog is written to allow this but forgot one check when updating the button text which I added.

Comment (wwylele) [reply src/citra_qt/configuration/configure_input.cpp]: @RicBent Sorry my bad. Didn't let github unfold all code for me.

File: src/core/settings.h (hunk 1 above)
Comment (RicBent) [reply src/core/settings.h]: Fixed with latest commit

**Comment by RicBent** [comment]
I was told that it is not really necessary to make the UI look quite a bit worse just for this button not many would use. It would look like this with the added button:

image

**Comment by MerryMage** [comment]
What are thoughts on implementing bit 13 as well, since that too comes from GPIO?

**Comment by RicBent** [comment]
I can implement that button too. Any name suggestions for that?
Also if I add another button the UI does look sane again.

**Comment by RicBent** [comment]
Does this look alright?

image

Here is a doc mentioning these GPIO pins btw: https://www.3dbrew.org/wiki/GPIO_Services#GPIO_Bits

**Comment by tgsm** [comment]
Should these new buttons be added to [movie playback](https://github.com/citra-emu/citra/blob/3989c17cb088b7e1531e326eaa88fa1fb0c862eb/src/core/movie.cpp#L152-L163)?

**Comment by RicBent** [comment]
Movie playback support has been added with latest commit.

File: src/citra_qt/configuration/configure_input.cpp [hunk 3]
```diff
@@ -273,7 +275,8 @@ void ConfigureInput::ClearAll() {
 
 void ConfigureInput::updateButtonLabels() {
     for (int button = 0; button < Settings::NativeButton::NumButtons; button++) {
-        button_map[button]->setText(ButtonToText(buttons_param[button]));
+        if (button_map[button])
+            button_map[button]->setText(ButtonToText(buttons_param[button]));
```
Comment (zhaowenlan1779) [inline src/citra_qt/configuration/configure_input.cpp]: is this needed anymore?

Comment (RicBent) [reply src/citra_qt/configuration/configure_input.cpp]: It is not but the entire dialog does nullptr checks and it rather looks like this one was missed.

**Pull request merged**